src/Storages/System/StorageSystemSettingsChanges.cpp

```diff
@@ -26,6 +26,7 @@ ColumnsDescription StorageSystemSettingsChanges::getColumnsDescription()
 
 void StorageSystemSettingsChanges::fillData(MutableColumns & res_columns, ContextPtr, const ActionsDAG::Node *, std::vector<UInt8>) const
 {
+    const auto & settings_changes_history = getSettingsChangesHistory();
```

Comment:
I don't understand why using `auto settings_changes_history` causes the current version's compatible parameters to be set to `previous_value`.